internal/reconciler/fake.go

```diff
@@ -61,15 +70,40 @@ func (c *FakeClient) GetCluster(clusterName string, configVersion int64) (*recon
 
 	existingCluster, exists := c.inventoryClusters[clusterName]
 	if !exists {
-		return &reconcilerApi.HTTPClusterResponse{}, fmt.Errorf("not found")
+		msg := stringKeys(c.inventoryClusters)
+		return &reconcilerApi.HTTPClusterResponse{}, fmt.Errorf("cluster not found, name=%s (%s)", clusterName, msg)
+	}
+	existingCluster.numberOfGetStatusCalls = existingCluster.numberOfGetStatusCalls - 1
+	if existingCluster.numberOfGetStatusCalls <= 0 {
+		existingCluster.numberOfGetStatusCalls = 0
 	}
 	state, exists := existingCluster.clusterStates[configVersion]
 	if !exists {
-		return &reconcilerApi.HTTPClusterResponse{}, fmt.Errorf("not found")
+		msg := intKeys(existingCluster.clusterStates)
+		return &reconcilerApi.HTTPClusterResponse{}, fmt.Errorf("cluster state not found, version=%d (%s)", configVersion, msg)
+	}
+	if existingCluster.numberOfGetStatusCalls == 0 && c.expectedStatus != nil {
+		state.Status = *c.expectedStatus
 	}
 	return state, nil
 }
 
+func intKeys(states map[int64]*reconcilerApi.HTTPClusterResponse) string {
```

Comment:
Do we just join all states from states map?

Reply:
I've added states